testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -112,12 +112,23 @@
 
 Then(/^I wait until I see the (VNC|spice) graphical console$/) do |type|
   repeat_until_timeout(message: "The #{type} graphical console didn't load") do
-    break unless has_xpath?('.//canvas')
+    break if find(:xpath, '//canvas')
+
+    # If the connection failed try reloading since the VM may not have been ready
+    if find(:xpath, "//*[contains(@class, \"modal-title\") and text() = '#{Failed to connect}']")
```

Comment:
`#{Failed to connect}` very strange. What's that? A variable?

Reply:
arf, no... it's a reminder of a previous step iteration: I'll make it a real string